### ISSUE TITLE
Release 2022.05.1

### ISSUE THREAD

**Issue by jsignell** [dask/community#245]
Apologies for opening this rather late in the day. Normally we would release 2022.05.1 tomorrow, but I heard from @jrbourbeau (who heard it from @crusaderky I thinkg) that there is a release blocker on distributed. I am proposing that we delay the release until Friday 2022-05-19. 

cc @jakirkham @quasiben @jsignell @fjetter @crusaderky

**Comment by crusaderky** [comment]
This is the blocker, ready for review and merge:
- https://github.com/dask/distributed/pull/6327

**Comment by jakirkham** [comment]
We also ran into this issue recently:

* https://github.com/dask/distributed/issues/6320

cc @pentschev

**Comment by jrbourbeau** [comment]
Checking in here for releasing tomorrow. It looks like @fjetter has a PR open (xref https://github.com/dask/distributed/pull/6363) that will close https://github.com/dask/distributed/issues/6320. Are there other PRs in `distributed` we want to make sure are merged before releasing? (cc @fjetter @crusaderky)

**Comment by crusaderky** [comment]
@jrbourbeau https://github.com/dask/distributed/issues/6305 is a regression introduced after the last release, and quite a nasty one IMHO. Not sure about @fjetter's timeline for it.

**Comment by jrbourbeau** [comment]
Okay, thanks for flagging @crusaderky. I'll check in tomorrow as it's well past @fjetter normal working time. @fjetter if you're able to provide any additional context around https://github.com/dask/distributed/issues/6305 that would be appreciated

**Comment by fjetter** [comment]
- https://github.com/dask/distributed/pull/6363 should be ready to merge to address https://github.com/dask/distributed/issues/6320 If nobody raises any further concern, I assume this fixes the upstream issues
- We are struggling a bit to close https://github.com/dask/distributed/issues/6305 However, I'm increasingly confident that this should not be flagged as a release blocker, see also https://github.com/dask/distributed/pull/6404 for more info

**Comment by jrbourbeau** [comment]
Thanks for the update and all your work on those issues @fjetter. @quasiben @pentschev @jakirkham could you provide some feedback on if https://github.com/dask/distributed/pull/6363 is good to go?

**Comment by jrbourbeau** [comment]
@gjoseph92 reviewed https://github.com/dask/distributed/pull/6363 and has some concerns https://github.com/dask/distributed/pull/6363#pullrequestreview-980540137. Given these concerns, that it's nearing EOD for US-based folks, and European-based folks are already done for the day, I'm going to hold off on releasing until someone like @fjetter @crusaderky can take a look at @gjoseph92's feedback. We don't have to wait another full week to release, but I would like to give some time for https://github.com/dask/distributed/pull/6363#pullrequestreview-980540137 to be looked at

**Comment by gjoseph92** [comment]
There is also https://github.com/dask/distributed/pull/6270#issuecomment-1133358073, which I think would be a blocker on its own as well.

**Comment by jakirkham** [comment]
It would be nice to see some new issues to track these things. Discussions and requests in closed issues/PRs are hard to track

**Comment by graingert** [comment]
a PR for just removing that stray breakpoint: https://github.com/dask/distributed/pull/6417

**Comment by fjetter** [comment]
A few notes about the process

1. I 100% agree with @jakirkham that we should open **new** tickets about new release blockers. As soon as something is in main, we require a new ticket. From what I understand reading through the comments, we have two blockers for which we have one PR comment and one ticket
- https://github.com/dask/distributed/issues/6407
- https://github.com/dask/distributed/pull/6363#pullrequestreview-980540137

Is there anything else?

2. I think for announcement tickets like "Release XY" we should keep the top-level comment up-to-date such that new readers can come to the issue and see what the current status is without going through the entire ticket itself. Currently, to understand where we are, one needs to read the entire ticket and carefully read every comment to find out what our current decision is. it's also a bit difficult to distinguish what belongs to the previous release attempt vs the current release attempt vs the next release attempt.

I opened https://github.com/dask/community/issues/247 to discuss

3. Most importantly, I think all owners and core maintainers should put an emphasize on fixing and/or reviewing release blockers asap. It took a very long time until a fix was actually proposed and with https://github.com/dask/distributed/pull/6363 it was difficult and slow to get feedback about whether this would fix the issue. Nobody actually reviewed the PR until very late on Friday, **after** the PR was merged. I felt comfortable with the proposed changes (https://github.com/dask/distributed/pull/6363#issuecomment-1133157875) and while likely not everybody will be able to spot content problems (like https://github.com/dask/distributed/pull/6363#pullrequestreview-980540137), everybody should be able to perform a sanity check to spot things like https://github.com/dask/distributed/pull/6417. That includes me as well, I should not even push code like this but mistakes happen and this is what a review is for. We should not merge PRs without any review.

**Comment by mrocklin** [comment]
If we revert https://github.com/dask/distributed/pull/6363 and https://github.com/dask/distributed/pull/6270 then can we safely release?

**Comment by gjoseph92** [comment]
@mrocklin yes, but I think https://github.com/dask/distributed/pull/6363 was considered a fix for https://github.com/dask/distributed/issues/6320, which was considered a blocker?

**Comment by jrbourbeau** [comment]
I think we can safely revert https://github.com/dask/distributed/pull/6270 or merge https://github.com/dask/distributed/pull/6408 which aims to address the security concerns (cc @jacobtomlinson for confirmation). https://github.com/dask/distributed/pull/6363 on the other hand was a fix for this issue https://github.com/dask/distributed/issues/6320 which, I believe, would break Dask-CUDA if we released without a fix. I'd like someone like @fjetter to weight in on @gjoseph92's review comments here https://github.com/dask/distributed/pull/6363#pullrequestreview-980540137 (at least the review comments that he marked with a ⚠️) to see if they constitute blocking the release and, if so, what changes we want to make to resolve the concerns that were raised

**Comment by jacobtomlinson** [comment]
I would prefer to go with the alternative fix that @gjoseph92 suggested in https://github.com/dask/distributed/pull/6408#issuecomment-1134800571.

**Comment by mrocklin** [comment]
> I would say should hold off on the release until @gjoseph92 has resolved his concerns and if we can help, we can prioritize the investigation.

If you all can jump in here that would also be welcome. I think that @gjoseph92 is trying to address a couple of things blocking the release today.

**Comment by gjoseph92** [comment]
@quasiben and I talked:
* @jacobtomlinson will handle https://github.com/dask/distributed/issues/6407 with the approach in https://github.com/dask/distributed/pull/6408#issuecomment-1134800571
* I will make a follow-up PR fixing https://github.com/dask/distributed/pull/6270
    * Switch `Worker.close` back to closing the nanny by default, instead of restarting it
    * Remove half-closed state in `Scheduler.handle_worker_status_change` https://github.com/dask/distributed/pull/6363#discussion_r878475185
    * I may also remove the `Worker.terminate` vs `Worker.close`; this might make more sense to handle as part of https://github.com/dask/distributed/issues/6387. TBD.
* @quasiben will confirm my changes still resolve https://github.com/dask/distributed/issues/6320

**Comment by jacobtomlinson** [comment]
I've raised https://github.com/dask/distributed/pull/6420

**Comment by jrbourbeau** [comment]
2022.05.1 is out on PyPI and conda-forge PRs are open

**Comment by jrbourbeau** [comment]
Thanks all!

**Comment by mrocklin** [comment]
So it sounds like "no", simply reverting would not solve the problem. Is
that correct?
